github actions: add auto-deploy steps for a new flux-accounting tag

cmoussa1 opened the pull request:
This PR looks to add CI steps for automatically creating and deploying a flux-accounting release when a new tag is created. I just looked at flux-core's `main.yml` for the auto-deploy steps, but am not sure if I copied everything I needed to. Let me know if I missed anything.

Fixes #154

grondo reviewed (approved):
Sorry I dropped the ball on reviewing this one!

This looks good to me on initial inspection! Sometimes it is difficult to find errors before the actual workflow runs on a tag though, but that's ok, problems can be identified and fixed after the fact!

cmoussa1 commented:
No problem @grondo! It wasn't an urgent PR or anything, but I appreciate you reviewing this.

> Sometimes it is difficult to find errors before the actual workflow runs on a tag though, but that's ok, problems can be identified and fixed after the fact!

That's what I figured, so no worries. I can always push up a fix if need be.

Thanks again for approving. Just rebased and force pushed to catch up after the other PR's that landed before this one, so I'll set MWP shortly.
